internal/controllers/rest_controller.go

```diff
@@ -76,19 +129,68 @@ func (rc RESTController) OIDCProviderCallback(w http.ResponseWriter, r *http.Req
 	}
 
 	// If the flow is secure, a JWT will be created...
-	jwt, err := rc.oidcService.CreateJWT(&token.CustomClaims)
+	jwt, err := rc.oidcService.Create(&token.CustomClaims)
 	if err != nil {
 		rc.logger.Error("an error occurred while creating a JWT. Error: " + err.Error())
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
-	// ...and then a JSON data to display as body content.
-	jsonBody, err := json.Marshal(jwt)
-	if err != nil {
-		rc.logger.Error("an error occurred while validating some claims. Error: " + err.Error())
-		return
+	// ...set a session cookie.
+	rc.setSessionCookie(w, r, "access_token", jwt.AccessToken)
+
+	rc.logger.Infof("Redirecting back to %s", rc.configuration.AuthServiceURL)
+	http.Redirect(w, r, rc.configuration.AuthServiceURL, http.StatusFound)
+}
+
+// Logout is responsible for revoking a token and deleting an authentication cookie.
+func (rc RESTController) Logout(w http.ResponseWriter, r *http.Request) {
+	// TODO: revoke token
+	rc.deleteCookie(w, "access_token")
+}
+
+// getCredentials is responsible for simply getting credentials.
+func (rc RESTController) getCredentials(r *http.Request) (string, error) {
+	rc.logger.Info("Get request credentials...")
+	for i, auth := range rc.credentials {
+		token, err := auth.Extract(r)
+		if err != nil {
+			if token == "" {
+				rc.logger.Debugf("[%d] credential error: %v", i, err)
+				continue
+			}
+			// if there's an error AND a token has been sent, we must send it to upstream
+			// generate an error response.
+			return token, err
+		}
+		// there's no error but somehow the token returned empty.
+		if token == "" {
+			continue
+		}
+		// credentials sent.
+		return token, nil
 	}
+	// back to start OIDC flow.
+	return "", nil
+}
```

Comment:
not the biggest fan.

returning `"", nil` is like going against all the cases that you shouldn't, is the case you never want (i would not, on my logic).
if you have an error, use it

you can have a cookie and an authorization header, if they are different, i think header one has preference, for example.
your case doesn't implement that

Reply:
good idea! 💡

Reply:
about the preference,
we can just change the position to header runs first and if there's a token, just return and that's it.

Reply:
please, see again and if you agree.